reframe/core/systems.py

```diff
@@ -194,85 +354,35 @@ def environment(self, name):
         return None
 
     @property
-    def arch(self):
-        '''The microarchitecture of this partition.
+    def processor(self):
+        '''The processor object of the current partition.
 
         .. versionadded:: 3.5
```

Comment:
Use semantic versioning conformat strings everywhere:

```suggestion
        .. versionadded:: 3.5.0
```